modules/storage/postgresql-single-server/main.bicep

```diff
@@ -209,31 +199,150 @@ resource postgresqlSingleServerConfig 'Microsoft.DBforPostgreSQL/servers/configu
 }]
 
 @batchSize(1)
-module postgresqlSingleServerRBAC 'modules/rbac.bicep' = [for (roleAssignment, index) in roleAssignments: {
-  name: 'postgresql-rbac-${uniqueString(deployment().name, location)}-${index}'
+module postgresServerRBAC 'modules/rbac.bicep' = [for (roleAssignment, index) in roleAssignments: {
+  dependsOn: [
+    postgresServer
+  ]
+  name: '${serverName}-rbac-${index}'
   params: {
-    description: contains(roleAssignment, 'description') ? roleAssignment.description : ''
+    description: roleAssignment.?description ?? ''
     principalIds: roleAssignment.principalIds
     roleDefinitionIdOrName: roleAssignment.roleDefinitionIdOrName
-    principalType: contains(roleAssignment, 'principalType') ? roleAssignment.principalType : ''
-    serverName: sqlServerName
+    principalType: roleAssignment.?principalType ?? ''
+    serverName: serverName
   }
 }]
 
-module postgresqlSingleServerPrivateEndpoint 'modules/privateEndpoint.bicep' = {
-  name: '${sqlServerName}-${uniqueString(deployment().name, location)}-private-endpoints'
+module postgresServerPrivateEndpoint 'modules/privateEndpoint.bicep' = {
+  name: '${serverName}-private-endpoints'
   params: {
     location: location
     privateEndpoints: varPrivateEndpoints
     tags: tags
   }
 }
 
+// ------ Diagnostics settings ------
+// May the diagnosticSettings become a module? How can I share the type cross module?
+@description('Provide postgres diagnostic settings properties.')
+param diagnosticSettingsProperties diagnosticSettingsPropertiesType = {}
+
+@description('Enable postgres diagnostic settings resource.')
+var enablePostgresDiagnosticSettings  = (empty(diagnosticSettingsProperties.?diagnosticReceivers.?workspaceId) && empty(diagnosticSettingsProperties.?diagnosticReceivers.?eventHub) && empty(diagnosticSettingsProperties.?diagnosticReceivers.?storageAccountId) && empty(diagnosticSettingsProperties.?diagnosticReceivers.?marketplacePartnerId)) ? false : true
+
+resource postgresDiagnosticSettings 'Microsoft.Insights/diagnosticSettings@2021-05-01-preview' = if (enablePostgresDiagnosticSettings) {
+  name: '${serverName}-diagnostic-settings'
+  properties: {
+    eventHubAuthorizationRuleId: diagnosticSettingsProperties.diagnosticReceivers.?eventHub.?EventHubAuthorizationRuleId ?? null
+    eventHubName:  diagnosticSettingsProperties.diagnosticReceivers.?eventHub.?EventHubName ?? null
+    logAnalyticsDestinationType: diagnosticSettingsProperties.diagnosticReceivers.?logAnalyticsDestinationType ?? null
+    logs: diagnosticSettingsProperties.?logs ?? null
```

Comment:
@jeskew , is using `.?` and `?? null` redundant?

Reply:
Yes is redundant. I notice that is required only on variables. If a variable has a conditional expression, and the result of that expression is `null` then you are going to get an error 
```
The template variable 'XXX' is not found. 
```

To solve this, the workaround is to set an empty string in the variable (instead of null) and use the coalesce operator
```
    paramA = XXX ?? null
```
In this way, you can set `null`